File: 351002/Yakhnovets/rest-task/src/writer/controller/writer.controller.ts
import {
  Body,
  Controller,
  Delete,
  Get,
  HttpCode,
  Param,
  ParseIntPipe,
  Post,
  Put,
} from '@nestjs/common';
import { WriterService } from '../service/writer.service';
import { WriterRequestTo } from '../dto/writer-request.to';
import { WriterResponseTo } from '../dto/writer-response.to';

@Controller('api/v1.0/writers')
export class WriterController {
  constructor(private readonly service: WriterService) {}

  @Get()
  getAll(): Promise<WriterResponseTo[]> {
    return this.service.getAll();
  }

  @Get(':id')
  getById(@Param('id', ParseIntPipe) id: number): Promise<WriterResponseTo> {
    return this.service.getById(id);
  }

  @Post()
  create(@Body() dto: WriterRequestTo): Promise<WriterResponseTo> {
    return this.service.create(dto); // 201 by default
  }

  @Put(':id')
  update(
    @Param('id', ParseIntPipe) id: number,
    @Body() dto: WriterRequestTo,
  ): Promise<WriterResponseTo> {
    return this.service.update(id, dto);
  }

  @Delete(':id')
  @HttpCode(204)
  delete(@Param('id', ParseIntPipe) id: number): Promise<void> {
    return this.service.delete(id);
  }
}

File: 351003/Suchok/publisher/src/main/java/by/liza/app/controller/WriterController.java
package by.liza.app.controller;

import by.liza.app.dto.request.WriterRequestTo;
import by.liza.app.dto.response.WriterResponseTo;
import by.liza.app.service.WriterService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/v1.0/writers")
@RequiredArgsConstructor
public class WriterController {

    private final WriterService writerService;

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public WriterResponseTo create(@Valid @RequestBody WriterRequestTo requestTo) {
        return writerService.create(requestTo);
    }

    @GetMapping("/{id}")
    @ResponseStatus(HttpStatus.OK)
    public WriterResponseTo getById(@PathVariable Long id) {
        return writerService.getById(id);
    }

    @GetMapping
    @ResponseStatus(HttpStatus.OK)
    public List<WriterResponseTo> getAll() {
        return writerService.getAll();
    }

    @PutMapping
    @ResponseStatus(HttpStatus.OK)
    public WriterResponseTo update(@Valid @RequestBody WriterRequestTo requestTo) {
        return writerService.update(requestTo);
    }

    @DeleteMapping("/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    public void delete(@PathVariable Long id) {
        writerService.delete(id);
    }
}